src/common/mfu_daos.c

```diff
@@ -2066,6 +2081,7 @@ static int serialize_recx_array(struct hdf5_args *hdf5,
             (*iod).iod_size  = size;
             (*iod).iod_nr    = 1;
             (*iod).iod_recxs = &recxs[i];
+            (*iod).iod_name = *akey;
```

Comment:
@daltonbohning in serialize_akeys this line:
   d_iov_set(&iod.iod_name, (void*)akey, strlen(akey));  
I think it needs to use akey_kds[j].kd_key_len instead of strlen. I am not sure that is the correct way to set the iod_name.

Reply:
@daltonbohning either way, I think the d_iov_set call needs to use kd_key_len instead of strlen

Reply:
@dsikich I think the reason dfs.c only uses d_iov_set is because in dfs.c, there isn't a daos_key_t instance for an akey.
I do see some examples of this way, such as here:
https://github.com/daos-stack/daos/blob/c9ab50cd1047f483c4b8720cda086ce0611ecf7a/src/object/srv_ec_aggregate.c#L528
This works because a daos_key_t is actually just a d_iov_t:
https://github.com/daos-stack/daos/blob/c9ab50cd1047f483c4b8720cda086ce0611ecf7a/src/include/daos_types.h#L122

But yes, we should probably use kd_key_len in that case.

Reply:
@daltonbohning thanks, yeah I saw that the key was really just a d_iov_t last night after writing this comment and looking up the types, but yeah I think we will just need to remove the strlen in the d_iov_set call